cli/src/torii/routing.rs

```diff
@@ -272,15 +273,35 @@ async fn handle_blocks_stream(sumeragi: Arc<Sumeragi>, mut stream: WebSocket) ->
         ))
         .await?;
 
+    let mut interval = tokio::time::interval(std::time::Duration::from_millis(10));
     loop {
-        let blocks = sumeragi.blocks_from_height(from_height.try_into().wrap_err("Failed to convert `from_height` into `usize`. You should consider upgrading to a 64-bit arch")?);
-        if blocks.is_empty() {
-            break;
+        tokio::select! {
+            // This branch catches `Close` and unexpected messages
+            closed = async {
+                while let Some(message) = stream.try_next().await? {
+                    if message.is_close() {
+                        return Ok(());
+                    }
+                    iroha_logger::warn!("Unexpected message received: {:?}", message);
+                }
+                eyre::bail!("Can't receive close message")
+            } => {
+                match closed {
+                    Ok(()) =>  {
+                        return stream.close().await.map_err(Into::into);
+                    }
+                    Err(err) => return Err(err)
+                }
+            }
+            // This branch sends blocks
+            _ = interval.tick() => {
+                let blocks = sumeragi.blocks_from_height(from_height.try_into().wrap_err("Failed to convert `from_height` into `usize`. You should consider upgrading to a 64-bit arch")?);
+                stream_blocks(&mut from_height, blocks, &mut stream).await?;
+            }
+            // Else branch to prevent panic
+            else => ()
         }
-        stream_blocks(&mut from_height, blocks, &mut stream).await?;
-        std::thread::sleep(std::time::Duration::from_millis(10));
```

Comment:
This statement + loop cause whole iroha to stuck, because of blocking operation in async context.

Reply:
Yea, good that we are replacing it.